test/server/media-controller.spec.js

```diff
@@ -0,0 +1,255 @@
+const app = require( '../../config/express' );
+const request = require( 'supertest' );
+const http = require( 'http' );
+
+/**
+ * Tests the request-filtering-agent to block SSRF attacks
+ * change testHTMLBody to the body of an html file that
+ * you are testing on. For the default, it says <im in.>
+ * and is hosted in testHTMLHost. 
+ */
+
+const testHTMLBody = 'im in.';
+const portHTML = 1234;
+const testHTMLHost = `http/localhost:${portHTML}`;
+const testHTMLMetaHost = `http/0.0.0.0:${portHTML}`;
+const localhost = '127.0.0.1';
+
+const requestURL = `/media/get/${testHTMLHost}`;
+const requestMetaURL = `/media/get/${testHTMLMetaHost}`;
+const server = http.createServer( function( req, res ) {
+    res.writeHead( 200, { 'Content-Type': 'text/plain' } );
+    res.end( testHTMLBody );
+} );
+
+describe( 'Testing request-filtering-agent', function() {
+
```

Comment:
Please add a test for a valid https resource, e.g. using one of these: https://www.w3.org/People/mimasa/test/imgformat/. Just checking for 200 status code is probably sufficient.

Reply:
And one very last thing... Would be good to test at least one blocked HEAD request as well.

Reply:
@MartijnR , for the HEAD request my understanding was that a HEAD request was made if `_isPrintView` is true, aka when there is a referrer with `?print=true` in the url?

For the https resouce it's giving me a 500 for this request: `http://localhost:8005/media/get/https/www.w3.org/People/mimasa/test/imgformat/img/w3c_home_animation.gif`. This is after I've added the proper agent for https in `media-controller`. Not sure if my url is just incorrect

Reply:
Yes, you're totally right about that! All should be good for HEAD requests. Sorry about that.

Reply:
@MartijnR Hopefully everything is in order with https://github.com/enketo/enketo-express/pull/150/commits/298845cfd3f40ca27f5f2a9c865a018df3a82ebc :crossed_fingers: